CMake/HighFiveConfig.cmake.in

```diff
@@ -1,3 +1,14 @@
+
+# Get HighFive targets
 if(EXISTS "${CMAKE_CURRENT_LIST_DIR}/HighFiveTargets.cmake")
   include("${CMAKE_CURRENT_LIST_DIR}/HighFiveTargets.cmake")
 endif()
+
+# Options when used from external projects. Keep defaults
+option(HIGHFIVE_PARALLEL_HDF5 "Enable Parallel HDF5 support" @HIGHFIVE_PARALLEL_HDF5@)
+option(USE_BOOST "Enable Boost Support" @USE_BOOST@)
```

Comment:
I'm not entirely sure how to invoke this options. If I'm not mistaken one cannot specify options with `find_package(...)`

Reply:
I'm not understanding. Does the comment have to do with the highlighted code?

Reply:
Yes. `HighFiveConfig.cmake.in` is evaluated when a user calls `find_package(HighFive)` (and not when a user uses the main `CMakeLists.txt`). In my understanding `option(...)` is a command-line option, thus relevant only in `CMakeLists.txt` and not in `HighFiveConfig.cmake.in`. Consequently the user will have no way to control this option, and will always be stuck with the default.

Reply:
I believe that's not true. CMake option() is pretty most the same as creating a cached var. But we should definitely test these scenarios.

Reply:
I'm getting this
```
CMake Warning (dev) at /Users/tdegeus/miniconda3/envs/hf/share/HighFive/CMake/HighFiveConfig.cmake:9 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'USE_BOOST'.
Call Stack (most recent call first):
  CMakeLists.txt:5 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
```
for 
```cmake
project(test)
set(USE_BOOST FALSE)
find_package(HighFive REQUIRED)
add_executable(test main.cpp)
target_link_libraries(test HighFive)
```

Reply:
Ok. Thanks for the test. CMake is Pedantic but it makes sense, but I believe the behavior is still correct, it just transformed the normal var to the cached var